File: soutLink-frontend/src/app/services/models/project-request-response.ts
/* tslint:disable */
/* eslint-disable */
import { StudentResponse } from '../models/student-response';
import { SupervisorProjectResponse } from '../models/supervisor-project-response';
export interface ProjectRequestResponse {
  date?: string;
  description?: string;
  id?: number;
  numberStudents?: number;
  projectTitle?: string;
  status?: 'PENDING' | 'REJECTED' | 'ACCEPTED';
  students?: Array<StudentResponse>;
  supervisor?: SupervisorProjectResponse;
}

File: soutLink-backend/src/main/java/com/soutLink/student/dto/ProjectRequestResponse.java
package com.soutLink.student.dto;

import com.soutLink.enums.ProjectRequestStatus;
import com.soutLink.supervisor.dto.StudentResponse;
import lombok.*;

import java.util.List;

@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
@Builder
public class ProjectRequestResponse {

    private Long id;
    private String projectTitle;
    private int numberStudents;
    private ProjectRequestStatus status;
    private SupervisorProjectResponse supervisor;
    private List<StudentResponse> students;
    private String description;
    private String date;
}
